### PR TITLE
add tests for missing aggregation

### PR DESCRIPTION
**Pull request by andyb-elastic**
Test with unmapped fields and using the missing parameter, which isn't
very useful with this aggregation but does work as expected. Also
includes yaml tests

For #42949

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/missing/MissingAggregatorTests.java [hunk 1]
```diff
@@ -28,81 +28,115 @@
...
+                    doc.add(new SortedNumericDocValuesField(fieldType.name(), randomLong()));
                 }
             },
             internalMissing -> {
                 assertEquals(internalMissing.getDocCount(), count.get());
                 count.set(0);
                 assertTrue(AggregationInspectionHelper.hasValue(internalMissing));
-            });
+            },
+            List.of(fieldType, anotherFieldType));
```
Comment (andyb-elastic) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/missing/MissingAggregatorTests.java]: Note: in some of the aggregator tests similar to this one (where there's a second field we expect to be absent sometimes), we don't pass a field type for the second field like above. If the test is missing the second field type, that field can't be read even if an aggregator was erroneously trying to collect it

**Review by polyfractal — COMMENTED** [review]
Test changes look good, thanks for beefing them up!  

Let's also add in the new "supported type" tests.  Here's a recently-merged example for `terms` agg: https://github.com/elastic/elasticsearch/blob/master/server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java#L127-L136

Missing agg supporting `missing`... I guess I'm not surprised but yeah, what a not-useful option.  I wonder if we should deprecate that in the future....

Edit:  Oh, let's add a test for scripting too, while we're at it?  Here's an example from the Avg tests: https://github.com/elastic/elasticsearch/blob/master/server/src/test/java/org/elasticsearch/search/aggregations/metrics/AvgAggregatorTests.java#L341

**Comment by andyb-elastic** [comment]
Thanks, I'll add tests for scripting and supported types. For some reason I thought the supported types tests weren't being run in master yet

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/missing/MissingAggregatorTests.java (hunk 1 above)
Comment (not-napoleon) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/missing/MissingAggregatorTests.java]: We should fix that, could you open a ticket so we don't forget please?

Comment (andyb-elastic) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/missing/MissingAggregatorTests.java]: I opened https://github.com/elastic/elasticsearch/issues/53250

**Comment by andyb-elastic** [comment]
@polyfractal added some scripting tests. Ended up changing the rest of the test cases to bring them more in line with the other agg tests and (I hope) make them a little more readable

**Review by polyfractal — APPROVED** [review]
LGTM!

**Comment by andyb-elastic** [comment]
Thanks!

**Pull request merged**